src/linear_algebra/quadratic_form.lean

```diff
@@ -303,3 +308,69 @@ by simp [discr, mul_left_comm, mul_comm]
 end discriminant
 
 end quadratic_form
+
+namespace quadratic_form
+
+variables {M₁ : Type*} {M₂ : Type*} {M₃ : Type*}
+variables [add_comm_group M₁] [add_comm_group M₂] [add_comm_group M₃]
+variables [module R M₁] [module R M₂] [module R M₃]
+
+/-- An isometry between two quadratic spaces `M₁, Q₁` and `M₂, Q₂` over a ring `R`,
+is a linear equivalence between `M₁` and `M₂` that commutes with the quadratic forms. -/
+@[nolint has_inhabited_instance] structure isometry
+  (Q₁ : quadratic_form R M₁) (Q₂ : quadratic_form R M₂) extends M₁ ≃ₗ[R] M₂ :=
+(map_app' : ∀ m, Q₂ (to_fun m) = Q₁ m)
+
+/-- Two quadratic forms over a ring `R` are equivalent
+if there exists an isometry between them:
+a linear equivalence that transforms one quadratic form into the other. -/
+def equiv (Q₁ : quadratic_form R M₁) (Q₂ : quadratic_form R M₂) := nonempty (Q₁.isometry Q₂)
```

Comment:
I don't think `equiv` is a great name because the global `equiv` is data, not a proposition.
How about `equivalent`?

Reply:
Done